src/renderer/views/Detail/Detail.tsx

```diff
@@ -0,0 +1,130 @@
+import React, { useState, useEffect } from 'react';
+import { useSelector } from 'react-redux';
+import { useParams } from 'react-router-dom';
+import { Section, Label, Input } from '../../components/Setting/Setting';
+import Button from '../../elements/Button/Button';
+import { RootState } from '../../store/reducers';
+import appStyles from '../../App.module.css';
+import * as LibraryActions from '../../store/actions/LibraryActions';
+import * as coverUtils from '../../../shared/lib/utils-cover';
+import Placeholder from '../../../images/assets/placeholder.png';
+import { parseDuration } from '../../lib/utils';
+
+import styles from './Detail.module.css';
+
+const Detail: React.FC = () => {
+  const { trackId } = useParams<{ trackId: string }>();
+  const tracks = useSelector((state: RootState) => state.library.tracks.library);
+  const [track, setTrack] = useState(undefined);
+  const [coverSrc, setCoverSrc] = useState(undefined);
+  const [data, setData] = useState({
+    title: '',
+    artist: '',
+    album: '',
+    genre: '',
+    duration: '',
+  });
+
+  const getCover = async (path: string) => {
```

Comment:
you can wrap all those functions in a `useCallback` (updateSong, handleSubmit, etc)